docs/adrs/024-remote-module-invocation.md

```diff
@@ -0,0 +1,280 @@
+runtime port. The optional `connection.protocol` config is used to select which protocol to send
+requests over, with the default being `grpc`. The `connection.tls` dictionary contains all information 
+related to TLS with `tls.enabled` controlling if the server is running SSL, `tls.ca` is the path to
+the CA file that the remote's certificate is signed by, and finally, `tls.cert` is the path to the
+MTLS client certificate, which will be sent with the request. The final connection config is `connection.options` 
+which defines a list of options to pass to either the HTTP or GRPC request; for an example of options,
+take a look at the [GRPC Channel options](https://grpc.github.io/grpc/core/group__grpc__arg__keys.html#details)
+
+
+Two additional optional fields help control what models this remote supports. The 
+`discover_models` setting is a boolean that controls if the finder should query the remote runtime
+to dynamically discover what models are loaded and their corresponding `module_id`'s. The
+`supported_models` config is a dictionary that contains a static mapping of model_paths to module_ids 
+that the remote supports. The `supported_models`  setting is required to add support for remotes that
+don't have a reflection api or ones that lazily load their models (like ModelMesh). 
+
+
+To help illustrate the above config, we included some pseudo python code to illustrate what happens 
+during model finding:
+```python
+def find_model(model_path: str)->RemoteModuleConfig:
+    # Check if model_path is in static mapping
+    if model_path in config.supported_models:
+        local_module = module_registry().get(config.supported_models[model_path])
+    
+    # Check if  model can be discovered dynamically
+    elif discover_models:
+        remote_model_mapping = gather_remote_model_map(config.connection)
+        local_module = remote_model_mapping.get(model_path)
+    
+    if not local_module:
+        raise CaikitCoreException("Model not found")
+    
+    # Construct config for use by the RemoteModelInitializer. This function is 
+    # described down below in the #RemoteModuleConfig section
+    return generate_config_for_module(local_module, config.connection, model_path)
+```
+
+## RemoteModelInitializer
+
+```yaml
+model_management:
+    initializers:
+        <initializer name>:
+            type: REMOTE
+```
+
+The proposed configuration for the RemoteModelInitializer is above. The initializer does not take in
+any global configuration settings, as the remote information will be passed in via the 
+RemoteModelFinder.  If a system is expected to have both local and remote models, consider using a 
+MultiModelInitializer to handle both use cases.
+
+To help illustrate how the RemoteModelInitializer would initialize a Module, we provided a snippet
+of pseudo Python code:
+```python3
+def init(model_config: RemoteModuleConfig)->ModuleBase:
+    # Construct empty RemoteModule Instance
+    @module(
```

Comment:
I think this cause problems if the IDs conflict with local modules that _do_ exist. There's nothing that requires the `Module` instance returned by a `ModelInitializer` to return an instance whose `MODULE_ID` matches the id of the config that got passed in.

Reply:
@gabe-l-hart so my idea for this was actually to use a modified module_id and not the exact same one e.g. `f"{original_module_id}-remote"`. That way you could have both Module's coexist

Reply:
Interesting, that would be nice. At some point, we had talked about enforcing a module ID to be a `uuid`, but that never got implemented, and personally I kind of like having some meaningful text in a module ID